api/APIv2.md

```diff
@@ -185,18 +185,20 @@ Return codes:
 * Success: `200`
 * Unknown Device: `400`
 * Controller is unavailable e.g., being upgraded: `503`
+* Attestation Failure, retry after a successful attestation: 403
 
 Request:
 
 The request MUST use the Device certificate to sign the protectedPayload in the AuthContainer. The senderCerthash MUST be set to the hash of the Device certificate.
 
 The request mime type MUST be "application/x-proto-binary".
-The request MUST have the body of a single protobuf message of type AuthContainer where the AuthBody is a protobuf message of type [config.ConfigRequest](./proto/config/devconfig.proto). The message should include the previous configHash from a previous ConfigResponse message to reduce network utilization when there is no configuration change.
+The request MUST have the body of a single protobuf message of type AuthContainer where the AuthBody is a protobuf message of type [config.ConfigRequest](./proto/config/devconfig.proto). The message should include the previous configHash from a previous ConfigResponse message to reduce network utilization when there is no configuration change. It should also provide a valid value in `integrityToken`, reflecting the value received from the Controller in the last successful remote attestation cycle.
 
 Response:
 
 The response mime type MUST be "application/x-proto-binary".
-The response MUST contain a single protobuf message of type AuthContainer where the AuthBody is a protobuf message of type [config.ConfigResponse](./proto/config/devconfig.proto). If the configHash is the same as in the ConfigRequest then the EdgeDevConfig should not be included. Otherwise the EdgeDevConfig will contain the entire configuration for the given Device.
+The response MUST contain a single protobuf message of type AuthContainer where the AuthBody is a protobuf message of type [config.ConfigResponse](./proto/config/devconfig.proto).
+If `integrityToken` value is not matching the copy stored in Controller, Controller should reject this config request with return code 403, indicating Attestation Failure. Device receiving such return code should complete a successful attestation cycle with Controller, get the latest value for `integrityToken`, and then resume config request with that latest value. If the configHash is the same as in the ConfigRequest then the EdgeDevConfig should not be included. Otherwise the EdgeDevConfig will contain the entire configuration for the given Device.
```

Comment:
Please make the "If the configHash ..." a separate line/paragraph. Otherwise it looks like it is related to integrityToken.

Reply:
Done